src/spec-lang/tc/typecheck.ts

```diff
@@ -902,6 +932,19 @@ function tcIdVariable(expr: SId, ctx: STypingCtx, typeEnv: TypeEnv): TypeNode |
     return defNode.parameters[bindingIdx][1];
 }
 
+export function tcIdImportUnitRef(
+    expr: SId,
+    ctx: STypingCtx,
+    typeEnv: TypeEnv
+): ImportRefType | undefined {
+    const curUnit = ctx[0] as SourceUnit;
+    const res = [...resolveAny(expr.name, curUnit, typeEnv.compilerVersion, true)].filter(
+        (x) => x instanceof ImportDirective
+    ) as ImportDirective[];
+
+    return res.length > 0 ? new ImportRefType(single(res)) : undefined;
+}
+
 export function tcId(expr: SId, ctx: STypingCtx, typeEnv: TypeEnv): TypeNode {
```

Comment:
Investigate if `tcId` could be further simplified into a single call to `resolveAny` and then a switch depending on the result. This might remove the need for `lookupVarDef`, `lookupFunDef`

Reply:
This fix would require further refactoring, and this PR is already growing quite large. Lets do it at a later time